android.Dockerfile

```diff
@@ -0,0 +1,157 @@
+# Download Debian-9 (Stretch) with openjdk-8 that is required for Android.
+ARG debian_snapshot=stretch-20191118
+FROM debian/snapshot:${debian_snapshot}
+
+RUN apt-get update && \
+    apt-get install -y \
+        bash-completion \
+        bison \
+        build-essential \
+        clang-format \
+        clang-tidy \
+        curl \
+        flex \
+        git \
+        libisl-dev \
+        libmpc-dev \
+        libmpfr-dev \
+        libncurses5 \
+        #libfl2 \
+        ocaml-nox \
+        #ocamlbuild \
+        openjdk-8-jdk \
+        python-dev \
+        python2.7-dev \
+        python3-dev \
+        rsync \
+        shellcheck \
+        texinfo \
+        vim \
+        wget \
+        xml2 \
+        zlib1g-dev
+
+# Install NodeJS.
+RUN curl -sL https://deb.nodesource.com/setup_12.x  | bash -
+RUN apt-get install -y nodejs
+
+RUN git --version
+RUN clang-format -version
+RUN shellcheck --version
+RUN node --version
+RUN npm --version
+
+# Install Bazel. 
+ARG bazel_version=1.2.1
+ARG bazel_sha=4bbb2718d451db5922223fda3aed3810c4ca50f431481e86a7fec4c585f18b1f
+ARG bazel_url=https://storage.googleapis.com/bazel-apt/pool/jdk1.8/b/bazel/bazel_${bazel_version}_amd64.deb
+RUN wget "${bazel_url}" -nv -o- -O bazel.deb && \
+    echo "${bazel_sha}  bazel.deb" > bazel.sha256 && \
+    sha256sum --check bazel.sha256 && \
+    apt-get install -y ./bazel.deb
+
+# Install prettier (via Node.js).
+# https://prettier.io/
+RUN npm install --global prettier
+RUN prettier --version
+
+# Install buildifier.
+ARG BAZEL_TOOLS_VERSION=0.29.0
+ARG BUILDIFIER_DIR=/usr/local/buildifier
+RUN mkdir -p $BUILDIFIER_DIR/bin
+RUN curl -L https://github.com/bazelbuild/buildtools/releases/download/${BAZEL_TOOLS_VERSION}/buildifier > $BUILDIFIER_DIR/bin/buildifier
+ENV PATH "$BUILDIFIER_DIR/bin:$PATH"
+RUN chmod +x $BUILDIFIER_DIR/bin/buildifier
+
+# Install Protobuf compiler.
+ARG PROTOBUF_VERSION=3.9.1
+ARG PROTOBUF_DIR=/usr/local/protobuf
+RUN curl -L https://github.com/protocolbuffers/protobuf/releases/download/v${PROTOBUF_VERSION}/protoc-${PROTOBUF_VERSION}-linux-x86_64.zip > /tmp/protobuf.zip
+RUN unzip /tmp/protobuf.zip -d $PROTOBUF_DIR
+ENV PATH "$PROTOBUF_DIR/bin:$PATH"
+RUN protoc --version
+
+# Install rustup.
+ARG RUSTUP_DIR=/usr/local/cargo
+ENV RUSTUP_HOME $RUSTUP_DIR
+ENV CARGO_HOME $RUSTUP_DIR
+RUN curl https://sh.rustup.rs -sSf > /tmp/rustup
+RUN sh /tmp/rustup -y --default-toolchain=none
+ENV PATH "$RUSTUP_DIR/bin:$PATH"
+RUN rustup --version
+RUN chmod a+rwx $RUSTUP_DIR
```

Comment:
Why do we need all those deps here?  was expecting this image is only used to build the android targets.

Reply:
Deleted unnecessary tools in Dockerfile.